src/Dfe.PlanTech.Domain/Questionnaire/Models/Section.cs

```diff
@@ -26,6 +26,31 @@ public class Section : ContentComponent, ISection
         if (string.IsNullOrEmpty(maturity) || !Enum.TryParse(maturity, out Maturity maturityResponse)) return null;
 
         return TryGetRecommendationForMaturity(maturityResponse);
+    }
 
+    public IEnumerable<QuestionWithAnswer> GetAttachedQuestions(IEnumerable<QuestionWithAnswer> responses)
+    {
+        var questionWithAnswerMap = responses.ToDictionary(questionWithAnswer => questionWithAnswer.QuestionRef,
+                                                                                 questionWithAnswer => questionWithAnswer);
+
+        Question? node = Questions[0];
+
+        while (node != null)
```

Comment:
Possibly check if node actually has an item in there.

Reply:
No need, as if it doesn't it'll just skip past this bit. It fulfils its purpose at that point then.